### ISSUE TITLE
show_pairs_scatter for prim

### ISSUE THREAD

**Issue by jpn--** [quaquel/EMAworkbench#63]
A question on the `show_pairs_scatter` feature when used with prim:  in the current implementation, the pairs plot figure is drawn with the current box limits overlaid on a scatterplot of the points.  The points plotted are all of the points, regardless if the box is the first box using all the data, or a subsequent box using a subset of the data.  It seems to me like it would be valuable to be able to plot only the `yi_initial` points for the current box, not the full dataset every time. Am I interpreting this right?  Or is there something about how prim is used that I am missing here?

If this is a reasonable thing, it's an easy change, which I can do and submit a PR.

**Comment by quaquel** [comment]
you are absolutely right. 

One more idea: do we want to be able to show multiple boxes as well (so different coloured squares?)

I fixed a bug yesterday in this code (is part of the master branch), so any pull request should start from this point. If I fix it today myself, I will let you know.

**Comment by jpn--** [comment]
I think multiple boxes on a single plot would be a valuable additional
feature.

Also, having some user selectable set of plotted dimensions... especially
if we expand to multiple boxes. I am seeing some instances in my work that
the total number of restricted dimensions can get large, especially if it’s
the union of all boxes. It would be easier to see and understand  the pairs
plots in some contexts if they could be generated only for a set of user
identified relevant dimensions.

**Comment by quaquel** [comment]
good point, I have a fix for the first issue which I am testing at the moment.

Showing multiple boxes should be a method on the prim algorithm rather than on an individual box. I assume you want the user selectable dimensions regardless of whether you are showing a single box or multiple boxes?

**Comment by jpn--** [comment]
Yes, user selectable dimensions regardless of whether you are showing a
single box or multiple boxes

**Comment by quaquel** [comment]
Finally time to work on this. Doing the multiple boxes becomes difficult with categorical data. Say, you have 3 boxes, with different categories being used as the box definition, visualising these restrictions using a single rectangle becomes tricky if not impossible.

It becomes impossible if box 1 uses categories A, B; box 2 uses A and C, while box 3 uses B and C. In this case it is not possible to order the categories such that all thee restrictions can be captures with separate rectangles. 

Admittedly this is a bit of a corner case. It does not happen often that you have 3 boxes. In fact, most of the time I can get by with just a single box.
